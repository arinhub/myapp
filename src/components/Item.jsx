import React from 'react';
import PropTypes from 'prop-types'
import { LazyLoadImage, trackWindowScroll} from 'react-lazy-load-image-component';
import 'react-lazy-load-image-component/src/effects/blur.css';


const Item = (props)  => {
  return (
    <div className={'item'}>
      <LazyLoadImage
      alt={props.name}
      height={"196px"}
      effect="blur"
      scrollPosition={props.scrollPosition}
      src={props['poster-image'] === 'posterthatismissing.jpg' ? `${process.env.PUBLIC_URL}/assets/placeholder_for_missing_posters.png` : `${process.env.PUBLIC_URL}/assets/${props['poster-image']}`}
      />
      <p className={'title'}>{props.name}</p>
    </div>
  )
};

Item.propTypes = {
  name: PropTypes.string.isRequired
}

export default trackWindowScroll(Item);