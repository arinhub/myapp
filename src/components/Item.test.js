import renderer from 'react-test-renderer';
import React from 'react';
import Item from './Item';

const mockDispatchFunction = jest.fn();

jest.mock('react-redux', () => {
  return {
    ...jest.requireActual('react-redux'),
    useSelector: jest.fn().mockImplementation(selector => selector()),
    useDispatch: () => {
      return mockDispatchFunction;
    },
  };
});

it('renders correctly', () => {
  const tree = renderer.create(
    <Item name="test"/>
  );
  expect(tree).toMatchSnapshot();
});
