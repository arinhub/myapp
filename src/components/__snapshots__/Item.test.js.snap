// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="item"
>
  <span
    className=" lazy-load-image-background blur"
    style={
      Object {
        "backgroundImage": "",
        "backgroundSize": "",
        "display": "inline-block",
        "height": "196px",
        "width": undefined,
      }
    }
  >
    <span
      className=""
      style={
        Object {
          "display": "inline-block",
          "height": "196px",
        }
      }
    />
  </span>
  <p
    className="title"
  >
    test
  </p>
</div>
`;
